From 4470faec975c55bffb53509d5d7b883c13ca5fab Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Thu, 15 Feb 2024 10:11:24 +0200
Subject: [PATCH] [SW-137122] gaudi3: remove tpm related files

Removed the tpm related files and offset, since
no more support in ext rom of gaudi2 and gaudi3
as well.

Change-Id: I02c5cf7f9802054ad1b22d78d8f44f5168628149
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index a35e518..7d05684 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -56,7 +56,6 @@
 	erom_secondary.o
 
 EROM_RECOVERY_DEPS := \
-	recovery_stubs.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/boot_desc.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/cmd_module.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/module_i2c_slave.o \
diff --git a/app/ext_rom/recovery_stubs.c b/app/ext_rom/recovery_stubs.c
deleted file mode 100644
index f29a792..0000000
--- a/app/ext_rom/recovery_stubs.c
+++ /dev/null
@@ -1,14 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright (C) 2021 HabanaLabs Ltd.
- * All Rights Reserved.
- *
- */
-
-#include <stdint.h>
-
-void preboot_tpm_auth(void) {}
-int tpm_pcr_extend(uint32_t image_index, uint8_t *hash)
-{
-	return 0;
-}
diff --git a/app/ext_rom/tpm_erom.lds.S b/app/ext_rom/tpm_erom.lds.S
deleted file mode 100644
index 5b8c438..0000000
--- a/app/ext_rom/tpm_erom.lds.S
+++ /dev/null
@@ -1,36 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0+ */
-/*
- * Copyright 2021 HabanaLabs, Ltd.
- * All Rights Reserved.
- */
-
-#include "common.h"
-
-MEMORY { .sram : ORIGIN = EXT_ROM_TPM_TEXT_BASE, LENGTH = EXT_ROM_TEXT_LENGTH }
-OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
-OUTPUT_ARCH(arc)
-
-ENTRY(_start)
-SECTIONS
-{
-	.text : ALIGN(4) {
-		crt0_common_erom.o (.text*)
-		* (.text*)
-	} > .sram
-
-	.bss : {
-		. = ALIGN(8);
-		__bss_start = . ;
-		*(.bss*)
-		__bss_end = . ;
-	} > .sram
-
-	.stack : {
-		. = ALIGN(8);
-		__stack = . ;
-		. = . + 512 ;
-		__stack_top = . ;
-	} > .sram
-
-	.note.gnu.build-id : { *(.note.gnu.build-id) } > .sram
-}
diff --git a/arch/gaudi2/arc/dev_ext_rom.h b/arch/gaudi2/arc/dev_ext_rom.h
index 6dcd79a..170c8a5 100644
--- a/arch/gaudi2/arc/dev_ext_rom.h
+++ b/arch/gaudi2/arc/dev_ext_rom.h
@@ -13,6 +13,5 @@
 /* No header bytes. Hence the text base is same as erom start address */
 #define EXT_ROM_PCIE_TEXT_BASE		(EXT_ROM_START)
 #define EXT_ROM_RECOVERY_TEXT_BASE	(SDRAM_BASE + 0x10000)
-#define EXT_ROM_TPM_TEXT_BASE		(EXT_ROM_START)
 
 #endif
diff --git a/arch/gaudi3/arc/dev_ext_rom.h b/arch/gaudi3/arc/dev_ext_rom.h
index 0972417..c29a668 100644
--- a/arch/gaudi3/arc/dev_ext_rom.h
+++ b/arch/gaudi3/arc/dev_ext_rom.h
@@ -12,6 +12,5 @@
 /* No header bytes. Hence the text base is same as erom start address */
 #define EXT_ROM_PCIE_TEXT_BASE		(EROM_START_OFFSET)
 #define EXT_ROM_RECOVERY_TEXT_BASE	(EROM_START_OFFSET)
-#define EXT_ROM_TPM_TEXT_BASE		(EROM_START_OFFSET)
 
 #endif
